### PR TITLE
doc: add `serialization` group to Doxygen

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This PR is the first of a series of PR for improving the documentation.

Here I group several serialization/deserialization format in a `Serialization` group, so they are easier to find for the end user. 

A `sys_serialization` group was defined under `sys` group.

The following modules where touched and added to the corresponding groups:

- cbor: sys_serialization
- base64: sys_serialization
- ubjson: sys_serialization
- cayenne_lpp: sys_serialization, pkg
- cn-cbor: sys_serialization, pkg
- jsnm: sys_serialization, pkg
- minmea: sys_serialization, pkg (I might add a Parsers section later, if applies)

### Testing procedure
`make doc`. Go to `System>Serialization` in the generated documentation and see how these components are grouped.

### Issues/PRs references
#8479

### CODE REVIEW

**Comment by aabadie** [comment]
@jia200x, this looks good in general but please rebase and fix the doxygen warning raised by the static tests.

File: sys/include/base64.h [hunk 1]
```diff
@@ -8,9 +8,7 @@
  */
 
 /**
- * @defgroup    sys_base64 base64 encoder decoder
```
Comment (smlng) [inline sys/include/base64.h]: Why these changes, IIRC I had this discussion before and we concluded to use `doc.txt` for group definition only when there is no proper header file. However, here we have these header files, so no need to introduce `doc.txt` files, too.

Comment (smlng) [reply sys/include/base64.h]: I mean for `sys_base64`, `sys_ubjson` and the like

Comment (jia200x) [reply sys/include/base64.h]: hmmmm, I could revert this change. I was adding doc.txt files because it's easier to write per-module documentation in a doc file rather than headers. Shall I revert?

Comment (jia200x) [reply sys/include/base64.h]: ping @smlng

Comment (smlng) [reply sys/include/base64.h]: as said I had this discussion in one of my own PRs and the consensus was to use header files where possible and limit the usage of `doc.txt` to cases where no header file is available for the module or group.

TL;DR: in this case I would suggest to revert, yes.

Comment (smlng) [reply sys/include/base64.h]: @miri64 IIRC we had this discussion, any thought/comments here?

**Comment by smlng** [comment]
btw. please rebase and resolve conflicts!

File: sys/include/base64.h (hunk 1 above)
Comment (miri64) [reply sys/include/base64.h]: I agree that adding `doc.txt` when not actually needed is not the way to go. The documentation stays with the code this way and isn't "hidden" in some file nobody ever really looks at. ;-)

Comment (jia200x) [reply sys/include/base64.h]: ok, I will revert them :)

**Comment by jia200x** [comment]
@smlng done!

File: sys/include/ubjson.h [hunk 2]
```diff
@@ -17,9 +17,9 @@
  */
 
 /**
- * @defgroup    sys_ubjson Universal Binary JSON
- * @ingroup     sys
- * @brief       A library to read and write UBJSON serialized data.
+ * @defgroup sys_ubjson  Universal Binary JSON deserializer
+ * @ingroup  sys_serialization
+ * @brief    Provides a BSON deserializer library to RIOT
```
Comment (miri64) [inline sys/include/ubjson.h]: BSON?

Comment (miri64) [reply sys/include/ubjson.h]: To clarify my question after a little bit of research: I don't think BSON and UBJSON are the same thing. [According to Wikipedia](https://en.wikipedia.org/wiki/UBJSON) UBJSON is aimed to be the successor of BSON, but that still wouldn't be the same thing.

Comment (jia200x) [reply sys/include/ubjson.h]: ok, I will differentiate them

**Comment by jia200x** [comment]
@miri64 done!

File: sys/include/ubjson.h [hunk 3]
```diff
@@ -17,9 +17,9 @@
  */
 
 /**
- * @defgroup    sys_ubjson Universal Binary JSON
- * @ingroup     sys
- * @brief       A library to read and write UBJSON serialized data.
+ * @defgroup sys_ubjson  Universal Binary JSON deserializer
+ * @ingroup  sys_serialization
+ * @brief    Provides a UBJSON deserializer library to RIOT
```
Comment (miri64) [inline sys/include/ubjson.h]: Still weird, that the original doc said, the lib is able to read and write UBJSON, but your new doc states its only able to deserialize (i.e. read)...

Comment (jia200x) [reply sys/include/ubjson.h]: You are right. Sorry

**Comment by jia200x** [comment]
@miri64 ok, fixed the `@defgroup` and `@brief`. Now it should be OK.

File: doc/doxygen/src/getting-started.md [hunk 4]
```diff
@@ -39,7 +39,7 @@ simplest way to compile and link an application with RIOT, is to set up a
 Makefile providing at least the following variables:
 
  * `APPLICATION`: should contain the (unique) name of your application
- * `BOARD`: specifies the platform the application should be build for by
+ * `BOARD`: specifies the platform the application should be built for by
```
Comment (miri64) [inline doc/doxygen/src/getting-started.md]: This was also fixed in https://github.com/RIOT-OS/RIOT/pull/10113 so you might not have rebased properly.

**Comment by jia200x** [comment]
@miri64 oh yes, there seemed to be a problem with the last push. Now it should be fixed

**Comment by jia200x** [comment]
yay :tada:

**Comment by miri64** [comment]
Wait -.- as far as I can see, lora-serilization is missing (#9863 was merged after this PR was opened). But I guess we can do that as a follow-up.

**Comment by jia200x** [comment]
> Wait -.- as far as I can see, lora-serilization is missing (#9863 was merged after this PR was opened). But I guess we can do that as a follow-up.

Sure, I can open the follow up

**Comment by miri64** [comment]
`xtimer_now64_continuity` caused false negatives again :-/

**Comment by jia200x** [comment]
> xtimer_now64_continuity caused false negatives again :-/

Now it passed, can we merge then? :)

**Comment by miri64** [comment]
Yupp let's do it. Even though the build is already a few days old we have to consider that this is *just* a doc update, so I think we don't need to retest ;-).
